.github/workflows/test.yaml

```diff
@@ -30,16 +30,58 @@ env:
 jobs:
   main-tests:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # We test against different versions of Python and Golang, but not
+          # against different versions of Node.
+          #
+          # - The Python version installed where dask-gateway and
+          #   dask-gateway-proxy is run by the end user matters, so we test
+          #   against all the versions we intend to support.
+          #
+          #   We could for example choose to test against the versions that
+          #   hasn't reached end of life yet: https://endoflife.date/python.
+          #
+          # - The Golang version that compiles
+          #   dask-gateway-server/dask-gateway-proxy bundled for the
+          #   dask-gateway-server Python package is the only thing that matters.
+          #   Due to that, we can test fewer versions of Golang.
+          #
+          #   We could for example choose to test against the versions that
+          #   hasn't reached end of life yet: https://endoflife.date/go.
+          #
+          #   FIXME: Golang 1.16 and above fails because our "go get" isn't
+          #          functioning as before. Golang 1.12 and below fails because
+          #          the stretchr/testify/assert library doesn't support golang
+          #          1.12 and below.
+          #
+          # - Node is a dependency for JupyterHub's configurable-http-proxy that
+          #   we test integration with. We can test against only one version and
+          #   that would be fine.
+          #
+          - python-version: "3.6"
```

Comment:
Dask as a whole dropped `3.6` recently, so I suggest we don't worry about supporting it here.

Reply:
Oh, okay hmmmm let's consider dropping support for Python 3.7 at a later time. For now, I added a commit that stops tests against Python 3.6 and requires Python 3.7+. Various tests rely on pre-build Dockerfiles that still use Python 3.7 etc so it would be  big bloat to make such change as part of this PR.